e2e-tests/src/test/emergency_finalizer.rs

```diff
@@ -24,3 +34,117 @@ async fn set_emergency_finalizer_test() -> anyhow::Result<()> {
     assert_eq!(current_finalizer, Some(public));
     Ok(())
 }
+
+async fn finalize_from_to<C: AsConnection + Sync>(
+    connection: &C,
+    from: BlockHash,
+    to: BlockHash,
+    finalizer: AlephKeyPair,
+) -> anyhow::Result<()> {
+    let from = connection.get_block_number(from).await?.unwrap();
+    let to = connection.get_block_number(to).await?.unwrap();
+
+    for i in (from + 1)..=to {
+        let hash = connection.get_block_hash(i).await?.unwrap();
+        connection.emergency_finalize(i, hash, finalizer).await?
+    }
+    Ok(())
+}
+
+async fn setup() -> anyhow::Result<AlephKeyPair> {
+    let config = setup_test();
+    let (finalizer, _seed) = AlephKeyPair::generate();
+    let connection = config.create_root_connection().await;
+
+    connection
+        .set_emergency_finalizer(finalizer.public().into(), TxStatus::Finalized)
+        .await?;
+    connection
+        .wait_for_n_sessions(2, BlockStatus::Finalized)
+        .await;
+
+    let accounts = get_validators_keys(config);
+
+    let new_validators: Vec<AccountId> = accounts
+        .iter()
+        .map(|pair| pair.signer().public().into())
+        .collect();
+
+    let seats = CommitteeSeats {
+        reserved_seats: 4,
+        non_reserved_seats: 0,
+        non_reserved_finality_seats: 0,
+    };
+
+    connection
+        .change_validators(
+            Some(new_validators[0..4].to_vec()),
+            Some(vec![]),
+            Some(seats.clone()),
+            TxStatus::InBlock,
+        )
+        .await?;
+    connection.wait_for_n_eras(1, BlockStatus::Finalized).await;
+    connection
+        .change_validators(
+            Some(new_validators[2..].to_vec()),
+            Some(vec![]),
+            Some(seats),
+            TxStatus::InBlock,
+        )
+        .await?;
+    disable_validators(&[0, 1]).await?;
+
+    Ok(finalizer)
+}
+
+/// Tests emergency finalizer. Runs on 6 nodes 0-6.
```

Comment:
7?

Reply:
oh, I get it - in that case, please write:
* Runs on 6 nodes 0-5
* setup 0-3
* setup 2-5
* disable 0-1 (this is ok)

Reply:
good catch :P fixed